### ISSUE TITLE
CUDA warning

### ISSUE THREAD

**Issue by rppawlo** [DARMA-tasking/vt#1385]
**Describe the bug**
On cuda builds, we get the warning below repeated for every object that includes vt headers:
```
../TPL/vt/src/vt/trace/trace_common.h(59): warning: integer conversion resulted in a change of sign
```

Looking at the code, a negative value is assigned to an unsigned type.
```
using TraceEntrySeqType   = uint32_t;

static constexpr TraceEntryIDType const no_trace_entry_id = 0;
static constexpr TraceEntrySeqType const no_trace_entry_seq = -1;
```

**To Reproduce**
build on cuda

**Expected behavior**
no warning :)

**Screenshots**

**Platform (please complete the following information):**
 - Supercomputer or desktop [e.g., Trinity, Sierra)
 - Compiler environment [e.g., clang-3.9, gcc8]
 - Libraries [e.g., mpich-2, openmpi)
 - Commit ID or release of VT [e.g., vt-1.0.0]

**Additional context**

**Comment by jstrzebonski** [comment]
That is already fixed on develop by commit https://github.com/DARMA-tasking/vt/commit/65c13f81518dd0dcde57ca3ce433baad7468c6c3
Maybe it would be worth porting it to the "release" version @lifflander ?

**Comment by rppawlo** [comment]
So how do releases work for vt? That fix looks like it was committed in June of 2020. Seems like a simple change that should have been in release 1.0.1 from 6 days ago. Is develop branch not merged into release regularly?

**Comment by PhilMiller** [comment]
> So how do releases work for vt? That fix looks like it was committed in June of 2020. Seems like a simple change that should have been in release 1.0.1 from 6 days ago. Is develop branch not merged into release regularly?

Our releases work on a "stable branch" model, not a "promotion" model. We make frequent enough changes on develop that would require application source changes or mess with various compilers that we found constant updates to be burdensome at the time the choice was made.

**Comment by cz4rs** [comment]
@rppawlo 
Thanks for the report. This will be included in `1.0.2` release and was already merged into release candidate branch.

**Comment by cz4rs** [comment]
closing manually after PR has been merged
